src/ga_session.cpp

```diff
@@ -3702,6 +3702,76 @@ namespace sdk {
         return sign_ga_transaction(*this, details);
     }
 
+    nlohmann::json ga_session::sign_psbt(const nlohmann::json& details)
+    {
+        nlohmann::json result = details;
+        const std::string txhex = extract_tx_from_psbt(details.at("psbt"));
+        const nlohmann::json tx_details = { { "transaction", txhex } };
+        const auto flags = tx_flags(m_net_params.is_liquid());
+        wally_tx_ptr tx = tx_from_hex(txhex, flags);
+
+        // Clear utxos and fill it with the one that will be signed
+        std::vector<nlohmann::json> inputs;
+        inputs.reserve(tx->num_inputs);
+        for (size_t i = 0; i < tx->num_inputs; ++i) {
+            bool found = false;
+            const std::string txhash_hex = b2h_rev(tx->inputs[i].txhash);
+            uint32_t vout = tx->inputs[i].index;
+            for (auto& utxo : result.at("utxos")) {
+                if (utxo.at("txhash") == txhash_hex && utxo.at("pt_idx") == vout) {
+                    // Populate prevout_script, which is needed for signing,
+                    // but not returned by get_unspent_outputs
+                    utxo["prevout_script"] = b2h(output_script_from_utxo(utxo));
+                    inputs.emplace_back(utxo);
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                inputs.emplace_back(nlohmann::json::object());
+            }
+        }
+
+        // Returned utxos are the ones to be signed
+        result["utxos"].clear();
+        for (const auto& utxo : inputs) {
+            if (!utxo.empty()) {
+                result["utxos"].emplace_back(utxo);
+            }
+        }
+
+        if (result["utxos"].empty()) {
+            return result;
+        }
+
```

Comment:
you could use a bool to track whether any inputs were matched, clear the result utxos only, and return here if no matches. Then when you return result below, you can `std::move` the non-empty inputs into 'utxos` to avoid more copies.

Reply:
Done